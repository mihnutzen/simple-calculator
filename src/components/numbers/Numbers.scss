@import 'src/styles/variables';

.Numbers {
  display: grid;
  grid-template-columns: repeat($numbers-grid-columns, 1fr);
  grid-template-rows: repeat($numbers-grid-rows, $height);

  grid-gap: 1px;
  background-color: #666;

  > button {
    background: #CCC;
    line-height: $height;

    &:hover {
      opacity: 0.8;
    }
  }

  > button:nth-child(1)  {
    grid-column: 1;
    grid-row: $numbers-grid-rows;
    grid-column-end: span 2;
  }
}