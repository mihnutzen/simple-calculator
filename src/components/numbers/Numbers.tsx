import React from 'react';

import { NumOpt } from '../../types/calculator';

import './Numbers.scss';

interface NumbersProps {
  onNumber: (nr: NumOpt) => void;
}

const Numbers = ({ onNumber }: NumbersProps) => {
  const numbers = Array.from(Array(10).keys());

  return (
    <div className="Numbers">
        {numbers.map((nr) => <button onClick={() => onNumber(nr)} key={nr}>{nr}</button>)}
        <button onClick={() => onNumber('.')}>.</button>
    </div>
  );
}

export default Numbers;