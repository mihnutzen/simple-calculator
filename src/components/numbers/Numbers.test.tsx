import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Numbers from './Numbers';

describe('The Numbers component', () => {
  it('should render correctly', () => {
    const onNumberMock = jest.fn();

    const { asFragment } = render(<Numbers onNumber={onNumberMock} />);
    const firstRender = asFragment()

    expect(firstRender).toMatchSnapshot();
  });

  it('should trigger onNumber with payload', () => {
    const onNumberMock = jest.fn();

    const { getByText } = render(<Numbers onNumber={onNumberMock} />);

    fireEvent.click(getByText(/1/))

    expect(onNumberMock).toHaveBeenCalledWith(1);
  });
});
