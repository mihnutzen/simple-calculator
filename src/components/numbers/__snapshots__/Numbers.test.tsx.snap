// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Numbers component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="Numbers"
  >
    <button>
      0
    </button>
    <button>
      1
    </button>
    <button>
      2
    </button>
    <button>
      3
    </button>
    <button>
      4
    </button>
    <button>
      5
    </button>
    <button>
      6
    </button>
    <button>
      7
    </button>
    <button>
      8
    </button>
    <button>
      9
    </button>
    <button>
      .
    </button>
  </div>
</DocumentFragment>
`;
